### PR TITLE
Add MariaDB external service

### PR DESCRIPTION
**Pull request by ClerySem**

### CODE REVIEW

File: src/cmd/api/db/db.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	_ "github.com/go-sql-driver/mysql"
+	log "github.com/sirupsen/logrus"
+)
+
+type database struct {
+	Db *sql.DB
+}
+
+func OpenConn(user string, userPwd string, addr string, name string) (database, error) {
```
Comment (vlamy) [inline src/cmd/api/db/db.go:15]: Cloud be 

```suggestion
func OpenConn(user, userPwd, addr, name string) (database, error) {
```

File: src/cmd/api/controllers/video_upload_handler.go [hunk 2]
```diff
@@ -49,25 +53,36 @@ func (v VideoUploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 	if typeOk := isSupportedType(file); !typeOk {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
+	// Update database
+	videoModel := models.VideoModelUpload{
+		Title: title,
+	}
+	clientId, err := dao.PutVideo(v.MariadbClient, videoModel)
```
Comment (RaboliotLeGris) [inline src/cmd/api/controllers/video_upload_handler.go:67]: `clientId` -> `publicID` ?

File: src/cmd/api/controllers/video_upload_handler_test.go [hunk 3]
```diff
@@ -4,12 +4,14 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"log"
```
Comment (RaboliotLeGris) [inline src/cmd/api/controllers/video_upload_handler_test.go:7]: Not the good one

File: src/cmd/api/controllers/video_upload_handler_test.go [hunk 4]
```diff
@@ -100,15 +102,26 @@ func TestVideoUploadHandler(t *testing.T) {
...
 
+			// Mock database
+			db, mock, err := sqlmock.New()
+			if err != nil {
+				log.Fatal("Cannot mock database : ", err)
+			}
+			defer db.Close()
+
+			mock.ExpectExec("INSERT INTO videos").WithArgs(tt.giveTitle).WillReturnResult(sqlmock.NewResult(1, 1))
+			//mock.ExpectExec("INSERT INTO videos").WithArgs("", "", tt.giveTitle).WillReturnResult(sqlmock.NewResult(1, 1))
```
Comment (RaboliotLeGris) [inline src/cmd/api/controllers/video_upload_handler_test.go:113]: If we don't need it, delete it

File: src/cmd/api/controllers/video_upload_handler_test.go [hunk 5]
```diff
@@ -100,15 +102,26 @@ func TestVideoUploadHandler(t *testing.T) {
 			s3Client := clients.NewS3ClientDummy(nil, nil, tt.putObject, nil)
 			amqpClient := clients.NewAmqpClientDummy(nil, nil)
 
+			// Mock database
+			db, mock, err := sqlmock.New()
+			if err != nil {
```
Comment (RaboliotLeGris) [inline src/cmd/api/controllers/video_upload_handler_test.go:107]: use `assert.NoError` ?

File: src/cmd/api/controllers/videos_list_handler_test.go [hunk 6]
```diff
@@ -2,14 +2,15 @@ package controllers_test
 
 import (
 	"encoding/json"
+	"log"
```
Comment (RaboliotLeGris) [inline src/cmd/api/controllers/videos_list_handler_test.go:5]: Not the good package

File: src/cmd/api/db/dao/video.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	log "github.com/sirupsen/logrus"
+
+	"github.com/Sogilis/Voogle/src/cmd/api/db/models"
+)
+
+func PutVideo(db *sql.DB, video models.VideoModelUpload) (string, error) {
+	id := uuid.NewString()
+	clientId := uuid.NewString()
+
+	query := "INSERT INTO videos (id, client_id, title) VALUES ('" + id + "','" + clientId + "', ?);"
```
Comment (RaboliotLeGris) [inline src/cmd/api/db/dao/video.go:17]: Use parameters instead building the query like this.

+ We can certainly prepare the query

File: src/cmd/api/db/dao/video.go [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		log.Error("Error, can't know how many rows affected:", err)
+		return "", err
+	}
+
+	log.Infof("%d row(s) inserted", nbRowAff)
+	return clientId, nil
+}
+
+func GetVideos(db *sql.DB) ([]models.VideoModel, error) {
```
Comment (RaboliotLeGris) [inline src/cmd/api/db/dao/video.go:34]: Later, we will probably want to limit the quantity of listed video or paginate them. If we don't do it now, we should add an issue about it.

File: src/cmd/api/metrics/metrics.go [hunk 9]
```diff
@@ -8,13 +8,13 @@ import (
 
 func InitMetrics() {
 	if err := prometheus.Register(TotalRequests); err != nil {
-		log.Error("Unable to register metrics.TotalRequests prometheus")
+		log.Warning("Unable to register metrics.TotalRequests prometheus")
```
Comment (RaboliotLeGris) [inline src/cmd/api/metrics/metrics.go:11]: While you are at it, knowing why we have those error can be cool

**Comment by RaboliotLeGris** [comment]
How do we handle the videos that are already uploaded on the squarescale infra ?

**Review by RaboliotLeGris — CHANGES_REQUESTED** [review]
How do we handle the already uploaded videos ?

File: src/cmd/api/db/dao/video.go (hunk 8 above)
Comment (ClerySem) [reply src/cmd/api/db/dao/video.go:34]: This issue ? #157

File: src/cmd/api/metrics/metrics.go (hunk 9 above)
Comment (ClerySem) [reply src/cmd/api/metrics/metrics.go]: Fixed : https://github.com/prometheus/client_golang/issues/716

File: src/cmd/api/controllers/video_upload_handler.go [hunk 10]
```diff
@@ -10,15 +11,20 @@ import (
 	log "github.com/sirupsen/logrus"
 	"google.golang.org/protobuf/proto"
 
+	"github.com/Sogilis/Voogle/src/cmd/api/db/dao"
+	"github.com/Sogilis/Voogle/src/cmd/api/db/models"
```
Comment (vlamy) [inline src/cmd/api/controllers/video_upload_handler.go:15]: `dbModel`

File: src/cmd/api/db/models/video_model.go [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package models
+
+type VideoModel struct {
```
Comment (vlamy) [inline src/cmd/api/db/models/video_model.go:3]: `s/VideoModel/Video`

File: src/cmd/api/db/models/video_model.go [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+	Id         string
+	PublicId   string
+	Title      string
+	VState     string
+	LastUpdate string
+}
+
+type VideoModelUpload struct {
```
Comment (vlamy) [inline src/cmd/api/db/models/video_model.go:11]: `s/VideoModelUpload/Video`

**Review by vlamy — APPROVED** [review]

**Pull request merged**